chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1972 @@
+        unlock.append(rs)
+
+        # 200TX
+        rs = RadioSetting("200tx", "200TX", RadioSettingValueBoolean(
+            bool(_mem.int_200tx > 0)))
+        unlock.append(rs)
+
+        # 500TX
+        rs = RadioSetting("500tx", "500TX", RadioSettingValueBoolean(
+            bool(_mem.int_500tx > 0)))
+        unlock.append(rs)
+
+        # 350EN
+        rs = RadioSetting("350en", "350EN", RadioSettingValueBoolean(
+            bool(_mem.int_350en > 0)))
+        unlock.append(rs)
+
+        # SCREEN
+        rs = RadioSetting("screen", "SCREEN", RadioSettingValueBoolean(
+            bool(_mem.int_screen > 0)))
+        unlock.append(rs)
+
+        # readonly info
+        # Firmware
+        if self.FIRMWARE_VERSION == "":
+            firmware = "To get the firmware version please download"
+            "the image from the radio first"
+        else:
+            firmware = self.FIRMWARE_VERSION
+
+        val = RadioSettingValueString(0, 128, firmware)
+        val.set_mutable(False)
+        rs = RadioSetting("fw_ver", "Firmware Version", val)
+        roinfo.append(rs)
+
+        # Driver version
+        val = RadioSettingValueString(0, 128, DRIVER_VERSION)
+        val.set_mutable(False)
+        rs = RadioSetting("driver_ver", "Driver version", val)
+        roinfo.append(rs)
+
+        # No limits version for hacked firmware
+        val = RadioSettingValueBoolean(self.FIRMWARE_NOLIMITS)
+        val.set_mutable(False)
+        rs = RadioSetting("nolimits", "Limits disabled for modified firmware",
+                          val)
+        roinfo.append(rs)
+
+        return top
+
+    # Store details about a high-level memory to the memory map
+    # This is called when a user edits a memory in the UI
+    def set_memory(self, mem):
+        number = mem.number-1
+
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.channel[number]
+        _mem4 = self._memobj
+        # empty memory
+        if mem.empty:
+            _mem.set_raw("\xFF" * 16)
+            if number < 200:
+                _mem2 = self._memobj.channelname[number]
+                _mem2.set_raw("\xFF" * 16)
+                _mem4.channel_attributes[number].is_scanlist1 = 0
+                _mem4.channel_attributes[number].is_scanlist2 = 0
+                _mem4.channel_attributes[number].unknown1 = 0
+                _mem4.channel_attributes[number].unknown2 = 0
+                _mem4.channel_attributes[number].is_free = 1
+                _mem4.channel_attributes[number].band = 0x7
+            return mem
+
+        # clean the channel memory, restore some bits if it was used before
+        if _mem.get_raw()[0] == "\xff":
+            # this was an empty memory
+            _mem.set_raw("\x00" * 16)
+        else:
+            # this memory was't empty, save some bits that we don't know the
+            # meaning of, or that we don't support yet
+            prev_0a = ord(_mem.get_raw()[0x0a]) & SAVE_MASK_0A
```

Comment:
TODO
This is a leftover from old code. It works, however i will look into which field really needs to be saved, since most of them are just zeroes, and change it to using the unknownXXX values.

Regarding get_raw, i was just using the other drivers in chirp as an example, please do:
grep get_raw drivers/*py | | egrep -v '(get_raw_memory|asbytes)'
grep get_raw drivers/*py | grep asbytes

Reply:
Fixed the get_raw usage